### PR TITLE
timeline component

### PR DESCRIPTION
**Pull request by edwardscull**

### CODE REVIEW

**Comment by mikemonteith** [comment]
Not sure about the semantics of this.

"Request medication" is a heading but then it has no content below it.
Should be be looking at this being an ordered list instead?

image

**Comment by mikemonteith** [comment]
We are using colour to convey the past/present/future state. Do we need some markup for screen readers to do the same?

**Comment by davidhunter08** [comment]
> Not sure about the semantics of this.
> 
> "Request medication" is a heading but then it has no content below it. Should be be looking at this being an ordered list instead?
> 
> image
> Edit: I've just noticed that it _is_ an ordered list!

Good point @mikemonteith - in all the [example timelines I've seen across gov](https://github.com/nhsuk/nhsapp-frontend/issues/28), they use a heading, but I haven't seen one with a heading and no content beneath.

Maybe that example is a bad one (and a content problem?) and we need to change it? Also add guidance, eg: "it must have content beneath the heading".

**Comment by davidhunter08** [comment]
> We are using colour to convey the past/present/future state. Do we need some markup for screen readers to do the same?

I'm not sure. We can include this in the accessibility testing we're planning.

File: src/components/timeline/_timeline.scss [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    &--past {
+      &:before {
+        border-color: $color_nhsuk-blue;
+      }
+    }
+  }
+
+  &__badge {
```
Comment (mikemonteith) [inline src/components/timeline/_timeline.scss:54]: Is the timeline icon a badge? We have [the badge component](https://nhsuk.github.io/nhsapp-frontend/components/badge/) already so might make sense to come up with a different name for this.

File: src/components/timeline/_timeline.scss [hunk 2]
```diff
@@ -0,0 +1,97 @@
+/* ==========================================================================
```
Comment (mikemonteith) [inline src/components/timeline/_timeline.scss:1]: There are a few "magic numbers" in this file.
`margin-top: 4px;`, `margin-top: -6px;` etc.
Can we refactor to make these values computed from the size variables?

File: src/components/timeline/_timeline.scss [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  &__badge {
+    flex-shrink: 0;
+    height: $timeline-badge-size-mobile;
+    margin-left: - calc(($timeline-badge-size-mobile / 2) + ($timeline-border-width / 2));
+    margin-right: nhsuk-spacing(4);
+    margin-top: 4px;
+    width: $timeline-badge-size-mobile;
+    z-index: 1;
+
+    @include mq($from: tablet) {
```
Comment (mikemonteith) [inline src/components/timeline/_timeline.scss:63]: We could probably define a mixin to do the height, width and margin positioning for all 4 variations of the timeline icon (small & regular X mobile & tablet).

File: src/components/timeline/timeline.njk [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  {% set isPastItem = 'nhsapp-timeline__item--past' if item.isPastItem else '' %}
+  
+  <li class="nhsapp-timeline__item {{ isPastItem | trim }}">
+    {% if item.active or item.isPastItem %}
+      <svg class="nhsapp-timeline__badge" aria-hidden="true" xmlns="http://www.w3.org/2000/svg" width="28" height="28" viewBox="0 0 28 28" fill="none">
+        <circle cx="14" cy="14" r="13" transform="matrix(-1.19249e-08 -1 -1 1.19249e-08 28 28)" fill="#005EB8" stroke="#005EB8" stroke-width="2"/>
+      </svg>
+    {% else %}
+      <svg class="nhsapp-timeline__badge nhsapp-timeline__badge--small" aria-hidden="true" xmlns="http://www.w3.org/2000/svg" width="14" height="14" viewBox="0 0 14 14">
+        <circle cx="7" cy="7" r="6" transform="matrix(-1.19249e-08 -1 -1 1.19249e-08 14 14)" fill="white" stroke="#AEB7BD" stroke-width="2"/>
```
Comment (mikemonteith) [inline src/components/timeline/timeline.njk:13]: The `transform` attribute can be deleted from the SVGs in this file

File: src/components/timeline/_timeline.scss (hunk 3 above)
Comment (edwardscull) [reply src/components/timeline/_timeline.scss]: updated @mikemonteith

File: src/components/timeline/_timeline.scss (hunk 1 above)
Comment (edwardscull) [reply src/components/timeline/_timeline.scss:86]: As discussed offline, will need @davidhunter08 input

File: src/components/timeline/_timeline.scss (hunk 2 above)
Comment (edwardscull) [reply src/components/timeline/_timeline.scss]: As discussed offline, will need @davidhunter08 input

**Comment by davidhunter08** [comment]
Screenshot 2024-07-02 at 15 33 37

Looking at the HTML output makes me wonder if we can refactor the CSS to make it a bit cleaner. Possibly use a 'background: node-blue.svg` on the `li:before`?

What do you think @edwardscull @mikemonteith ?

**Comment by edwardscull** [comment]
> Screenshot 2024-07-02 at 15 33 37
> Looking at the HTML output makes me wonder if we can refactor the CSS to make it a bit cleaner. Possibly use a 'background: node-blue.svg`on the`li:before`?
> 
> What do you think @edwardscull @mikemonteith ?

Yes that would be cleaner for sure. It's an image that probably makes sense to be a background image.

**Review by davidhunter08 — APPROVED** [review]

**Pull request merged**